Cargo.toml

```diff
@@ -12,3 +12,7 @@ members = [
 [profile.release]
 lto = true
 opt-level = 'z'
+overflow-checks = true
```

Comment:
Ha! I was like, where is the code change - I just see tests? That makes sense.

Reply:
I was thinking about the same, however currently we are bulding our binary for testing in debug mode, while the tests I wrote require a binary built in release mode, so we have 2 options

1. build it only in release mode and make sure the debug assertions are not removed from the release binary (there is a flag for that). However the stack trace might look funny if there is any error during the tests.
2. build it both in debug mode and release mode, and retest both binary. This would double the CI time.

Even if we add those tests to CI, it wouldn't matter much, because cargo tests always use the 'bench' profile and not the release profile (even if the binary is built with 
`cargo test --release`) That means we would never be able to catch if someone remove the `overflow-checks = true` from the release profile.

[profile.release]
overflow-checks = true

[profile.bench]
overflow-checks = true

Reply:
@qti3e 
I actually have a question here regarding our CI:
```
      - name: Build
        run: node build.js

      - name: Cargo Test
        uses: actions-rs/cargo@v1
        with:
          command: test
          args: --all-features --no-fail-fast
        env:
          CARGO_INCREMENTAL: '0'
          RUSTFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests'
          RUSTDOCFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests'

```

So by running node build.js we make a release binary and by runnig the cargo test command we also make a debug binary. Any reason why we generate the release binary? Do we plan to upload it to some repo, so later we can deploy it (would be a great thing I think, so we don't have to manually build XTC in our computer)

Reply:
> @qti3e
> I actually have a question here regarding our CI:
> 
> ```
>       - name: Build
>         run: node build.js
> 
>       - name: Cargo Test
>         uses: actions-rs/cargo@v1
>         with:
>           command: test
>           args: --all-features --no-fail-fast
>         env:
>           CARGO_INCREMENTAL: '0'
>           RUSTFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests'
>           RUSTDOCFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests'
> ```
> 
> So by running node build.js we make a release binary, and by runnig the cargo test command we also make a debug binary. Any reason why we generate the release binary? Do we plan to upload it to some repo, so later we can deploy it (would be a great thing I think, so we don't have to manually build XTC on our computer). It is just a slight improvemnet though.

That sounds like they don't intend for you to test this but assume that rust works. I am happy with that if you are?

Do the tests you have added run then, it is just that they are under the `bench` profile, so will always `panic`?

Reply:
> I was thinking about the same, however currently we are bulding our binary for testing in debug mode, while the tests I wrote require a binary built in release mode, so we have 2 options
> 
>     1. build all tests only in release mode, and make sure the debug assertions are not removed from the release binary (there is a flag for that). That way we need to run all tests only once. However the stack trace might look funny if there is any error during the tests.
> 
>     2. build all tests both in debug mode and release mode, and retest both binary. This would double the CI time.
> 
> 
> Even if we add those overflow tests to CI, it wouldn't matter much, because cargo tests always use the 'bench' profile and not the release profile (even if the binary is built with
> `cargo test --release` which produces release binary) That means we would never be able to catch if someone remove the `overflow-checks = true` from the release profile.
> 
> ```toml
> [profile.release]
> overflow-checks = true
> 
> [profile.bench]
> overflow-checks = true 
> ```


That sounds like they don't intend for you to test this but assume that rust works. I am happy with that if you are?

Do the tests you have added run then, it is just that they are under the `bench` profile, so will always `panic`?

Reply:
I just manually tested with `cargo test --release` which uses the the `bench` profile and it paniced as expected.